src/main/java/org/terasology/thirst/ThirstAuthoritySystem.java

```diff
@@ -63,24 +63,27 @@
     @ReceiveEvent
     public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player,
                               ThirstComponent thirst) {
-        thirst.lastCalculatedWater = thirst.maxWaterCapacity;
-        thirst.lastCalculationTime = time.getGameTimeInMs();
-        player.saveComponent(thirst);
+        resetThirst(player, thirst);
     }
 
     /**
-     * Updates the thirst component of an entity one last time before it is removed. Called when the entity is being
-     * removed.
+     * Initialize thirst attributes for a respawned player. Called when a player is respawned.
      *
-     * @param event  the event corresponding to the removal of an entity
-     * @param entity the entity being removed
-     * @param thirst the thirst component associated with the entity
+     * @param event  the event corresponding to the respawning of the player
+     * @param player a reference to the player entity
+     * @param thirst the player's thirst component (to be initialized)
      */
     @ReceiveEvent
-    public void beforeRemoval(BeforeDeactivateComponent event, EntityRef entity, ThirstComponent thirst) {
-        thirst.lastCalculatedWater = ThirstUtils.getThirstForEntity(entity);
+    public void onPlayerRespawn(OnPlayerRespawnedEvent event, EntityRef player,
+                                ThirstComponent thirst) {
+        logger.info("mark");
+        resetThirst(player, thirst);
+    }
+
+    private void resetThirst(EntityRef player, ThirstComponent thirst) {
```

Comment:
Would it make sense to expose the `resetThirst` functionality as event + handler? That way, another module could modify the behavior in case a player respawns after death. For instance, start only with half the thirst meter in general, and only respawn with full thirst meter if you have some item equipped.